CMakeLists.txt

```diff
@@ -2,7 +2,7 @@
 ############################ BASE ######################################
 
 cmake_minimum_required (VERSION 3.20 FATAL_ERROR)
-project(OpenTurbine CXX)
+project(OpenTurbine VERSION 23.0.1 LANGUAGES CXX)
```

Comment:
We should probably start out with something like version 0.1.0. I've tagged the initial commit to v0.0.0 (as in, I added a git-tag). Also, keep in mind that the version is handled through https://github.com/Exawind/openturbine/blob/main/cmake/GetGitRevisionDescription.cmake so you'll have to keep these two in sync.

Reply:
@rafmudaf I really like the idea of using a git revision as the version, this is the most precise way to tie a run of the code back to the actual code that was built.
It would be nice if cmake could place the git-revision-version in a macro in a generated header or something like that, so that code could then either display it, or a calling-code could display it, etc.
Also, would it be possible for it to be displayed by cmake when we build openturbine?

Reply:
@alanw0 Good point, I've created a pull request with this change in #11. As for the generated header with the git version, OpenTurbine is currently set up in this way following AMR Wind. The main CMakeList calls the cmake macro `generate_git_info` in [openturbine-utils.cmake](https://github.com/Exawind/openturbine/blob/main/cmake/openturbine-utils.cmake). That macro uses a few routines in [GetGitRevisionDescription.cmake](https://github.com/Exawind/openturbine/blob/main/cmake/GetGitRevisionDescription.cmake) to parse the git-status. Ultimately, OpenTurbineVersion.H is generated based on the results from [OpenTurbineVersion.H.in](https://github.com/Exawind/openturbine/blob/main/cmake/OpenTurbineVersion.H.in) in the [src/CMakeLists.txt](https://github.com/Exawind/openturbine/blob/main/src/CMakeLists.txt#L31). This header is currently only used to display the version info in the CLI and unit test driver:

```
$ ./openturbine -h
==============================================================================
                OpenTurbine (https://github.com/exawind/openturbine)

  OpenTurbine version :: v0.0.0-37-gcef9bff-DIRTY
  OpenTurbine Git SHA :: cef9bffc11001faba73520580ba129bc5d3a8ef7-DIRTY
  Exec. time       :: Thu Feb  9 13:56:51 2023

  No additional third-party libraries enabled

           This software is released under the MIT License.                
 See https://github.com/Exawind/openturbine/blob/main/LICENSE for details. 
------------------------------------------------------------------------------
```

Reply:
ok, great, sounds good.
I'm currently working on building openturbine myself, had to take a detour to clone/build/install kokkos...
Then I'll be much more familiar with all this stuff.